### ISSUE TITLE
Custom filtering rules are not applied at all times.

### ISSUE THREAD

**Issue by Tjezee** [AdguardTeam/AdGuardHome#2131]
* **Version of AdGuard Home server:**
 Version: 0.103.3
* **Device model:**
Raspberry Pi 3B
* **Operating system and version:**
Raspian OS current 32bit

### Expected Behavior
Custom filtering rules are applied at all times

### Actual Behavior
I have to press "apply" on the ' Custom filtering rules ' page again for them to register correctly

### Additional Information
I'm not sure when it happens but one of my blocklists filter a login button for some reason so I've whitelisted the
domain in question. This site gets accessed maybe once every two/three weeks so I'm not sure when the problem starts,
after a device restart or an update or what. After I press apply it works as intended again without changing anything else.

**Comment by ameshkov** [comment]
I cannot reproduce this.

Here's what I am doing:

1. Add `||example.org^` to custom filtering rules
2. Press "Apply"
3. Run `nslookup example.org AGH_IP_ADDRESS` (where AGH_IP_ADDRESS is the IP address of my AGH instance)
4. Check that the domain is blocked.

The only idea I have is that you have many blocklists installed and reloading them takes a lot of time. Could it be?

**Comment by Tjezee** [comment]
Hey, thanks for taking the time to reply.
I have 10 lists installed, 6 of those are rather small and very specific, 2 of them are the default and 1 bigger one
with a total of 450k in the list. I've ran an inferior adblock with over 3 million at some point so I doubt it is stressed with this amount.

The problem is indeed quite hard to diagnose as it works right after you press "Apply" so it comes as no surprise to me that it
works when you do the test. Something happens after a period of time which makes the custom list not work anymore.
It's like the blocklists get renewed automatically and applied but the custom list doesn't get applied afterwards.

**Comment by ameshkov** [comment]
Could you please try reproducing it with the latest beta version?

There were some relevant changes to the filters update code, maybe that's why I can't reproduce it.

**Comment by Tjezee** [comment]
Thanks for your suggestion! I'm now running on : Version: 0.104.0-beta1
Everything works for now and will report back if the problem reoccurs after a while.
Cheers for your efforts.

**Comment by Tjezee** [comment]
Greetings,

I have found the issue and AdGuard Home worked perfectly after all. The site in question apparently embeds the login bar from another website for some reason. This website was blocked in a list somewhere so when it could not contact the other site it just refused to show the login bar. Now that I've unblocked the target domain everything works just fine.

Thanks for your time and have a lovely day.
